src/server/prepare.js

```diff
@@ -68,6 +86,110 @@ function execCordovaPrepare() {
     return deferred.promise;
 }
 
+function shouldPrepare(currentState) {
+    // We should prepare if we don't have any info on a previous prepare for the current platform, or if there is a
+    // difference in the list of installed plugins, the merges files or the www files.
+    var currentPlatform = config.platform;
+    var previousState = previousPrepareStates[currentPlatform];
+
+    if (!previousState) {
+        return true;
+    }
+
+    currentState = currentState || getProjectState();
+
+    // Use JSON stringification to compare states. This works because a state is an object that only contains
+    // serializable values, and because the glob module, which is used to read directories recursively, is
+    // deterministic in the order of the files it returns.
+    var pluginsAreTheSame = JSON.stringify(currentState.pluginList) === JSON.stringify(previousState.pluginList);
+    var filesAreTheSame = JSON.stringify(currentState.files) === JSON.stringify(previousState.files);
+
+
+    return !pluginsAreTheSame || !filesAreTheSame;
+}
+
+function shouldInitPlugins(currentState) {
+    // We should init plugins if we don't have any info on a previous prepare for the current platform, or if there is
+    // a difference in the list of installed plugins or debug-host handlers.
+    var currentPlatform = config.platform;
+    var previousState = previousPrepareStates[currentPlatform];
+
+    if (!previousState) {
+        return true;
+    }
+
+    currentState = currentState || getProjectState();
+
+    // Use JSON stringification to compare states. This works because a state is an object that only contains
+    // serializable values, and because the glob module, which is used to read directories recursively, is
+    // deterministic in the order of the files it returns.
+    var pluginsAreTheSame = JSON.stringify(currentState.pluginList) === JSON.stringify(previousState.pluginList);
```

Comment:
Same here. Let's create a utility method that compares two lists and use it for all comparisons.